### ISSUE TITLE
Wies de Munck group b class 204a

### ISSUE THREAD

**Issue by WiesdeMunck** [cmda-bt/fe-course-20-21#18]
**Data Structures: Objects and Arrays**
what are arrays, how can you use them and how do they work?
Arrays are datasets that hold pieces of information, these pieces of information could be string numbers and booleans. This information will be between the brackets and commas in the array. You can call upon these pieces of data through calling them on their added number in the array. When doing this you should be aware that arrays start at the number zero this would be the number to call upon the first piece of data. The same stands for the upcoming pieces of date the second piece would be one and this would be three.

Object in arrays is a way to give each piece of date its own responsibility for managing its state. You could make you bigger pieces of date into smaller bite sized parts. This way the information can be kept local. And if more people start working on the same assignment they wont have worry about every single piece of information sinds it won't affect different parts

With arrays you can do more than just adding pieces of data. You can give added properties to your array to edit the given information. These properties can be things such as putting your information into uppercase or editing it in different ways to get to your solution.

But what if an array is massive and you want to look for a piece of content. For this you can use loops. For loops can make for loops to look through your array to look for pieces of content.

An important part with arrays is that you don't have to use them on them. If you were to combine arrays with objects and different functions and loops you could make more complex things such as calculations. This is something you would consider to be a method property that holds different given values.

**JavaScript and the Browser**
The web is a massive web of protocols, that you can join by having your computer connect to one machine and have it connect to a port to receive these protocols. Once you are connected to the web you will receive an ip address. This will be your online address consisting of a large sum of random numbers, But you can see it as your own online postal code.

When you type something in your address bar of your browser http protocols will connect to your address. and this connect will send your browser the information you will get to see on your screen. 

the information sent to your computer to form your screen will exist of html this is a markup language consisting of lines of code  surrounding the content you will get to see on your screen.  but html is not alone responsible for the browsers content html is the content but to give the content its own place and style you would be using a second language called css. Css will shape the content, give it color and edit things like the typeface without css your browser would be one long list of black and white.

We now have html and css but there is one more large contender in this picture and this is javascript. Javascript will allow the browser to interact with the user. These things would include  changing shapes adding new content or sending information around the page

These three will make your browser look the way it does. The browser in interpet the code and makes the page. But this is still a small thing to add. there are multiple browsers and they will all interpret the code in a similar way but not the same this is the reason why sites may look different on multiple browsers.

**The Document Object Model**
When making a web page out of html. this should be made in a structure. This is called the document object model or dom for short. this is the way hey you put in your html.  this structure will have different branches depending on the html. 

one important thing about the dom is that it can be manipulated. this is something you would do with javascript. you can change the way your original structure of your html. you can add change and remove things with javascript.

to start manipulating your dom you first to select the party you would like to have edited. you can do this with selectors such as getElementbyID() or QuerySelector() the id is for when you have given a specific html element a id. The QuerySelector is for when you would like the call on the direct elements or class. Once an element is selected you can now edit it through javascript itself by removing or adding more html, animate with css  still in javascript but you can change add and remove the css  for the selected element.

This is the interactive part of the dom and opens a whole new element to your webpage. Now when you interact with a button the structure and style of the page can be edited to suit your desires. This can be added to check your form if everything that is filled in is correct or if you can delete parts that were first used but not anymore now a certain ability is reached on your page. 

in short the dom is what makes up the original pagee but you can change it to what you need by changing the structure and css with javascript.
